website/content/docs/auth/jwt/oidc_providers.mdx

```diff
@@ -262,3 +262,148 @@ vault write auth/oidc/role/your_default_role \
 1. Save client ID and secret.
 
 Note your policy will need `oidc_scopes` to include `profile` to get a full profile ("[Fat Token](https://support.okta.com/help/s/article/Okta-Groups-or-Attribute-Missing-from-Id-Token)"). You will also need to configure bound audience along the lines of `"bound_audiences": ["api://default", "0a4........."]` if you are using the default authorization server.
+
+## Kubernetes
+
+Kubernetes can function as an OIDC provider such that Vault can validate its
+service account tokens using JWT/OIDC auth.
+
+-> **Note:** The JWT auth engine does **not** use Kubernetes' `TokenReview` API
+during authentication, and instead uses public key cryptography to verify the
+contents of JWTs. This means tokens that have been revoked by Kubernetes will
+still be considered valid by Vault until their expiry time. To mitigate this
+risk, use short TTLs for service account tokens or use
+[Kubernetes auth][/docs/auth/kubernetes] which _does_ use the `TokenReview` API.
+
+### Using service account issuer discovery
+
+Kubernetes cluster requirements:
+
+* `ServiceAccountIssuerDiscovery` feature gate enabled.
+   * Present from 1.18, defaults to true from 1.20.
+* Must use short-lived service account tokens when logging in.
+   * Tokens mounted into pods default to short-lived from 1.21.
+* kube-apiserver's `--service-account-issuer` flag is set to a URL that is
+  reachable from Vault. Public by default for most managed Kubernetes solutions.
+
+1. Ensure OIDC discovery URLs do not require authentication, as detailed
+   [here](k8s-sa-issuer-discovery):
+
+   ```bash
+   kubectl create clusterrolebinding oidc-reviewer  \
+      --clusterrole=system:service-account-issuer-discovery \
+      --group=system:unauthenticated
+   ```
+
+1. Enable and configure JWT auth in Vault.
+
+   1. If Vault is running in Kubernetes:
```

Comment:
I am missing the `ISSUER` variable here that is defined in the alternative step and used in the role configuration step, can this be added or a link to the [discovering the service account issuer](https://www.vaultproject.io/docs/auth/kubernetes#discovering-the-service-account-issuer) block?

Reply:
@benashz I did notice that the standard issuer is used here when writing to the configuration of the JWT auth.

However, if you would only follow the steps that you are supposed to follow for when Vault is running in the cluster: you end up not setting the "env" variable for `ISSUER`.
This variable is later on needed to supply a bound audience for the created role and if not setting this would mean that this audience will be empty thus creating a role that would be invalid and any login with the Kubernetes issuer would fail.

While this is up to the Vault administrator to have enough knowledge about the used audience, it would be nice to have this in both steps. As this could be mistaken very easily as this variable is set in the alternative step, but not in this step; which might come over as "done on purpose". 😃

Reply:
I've pulled getting the `ISSUER` out into a separate step to avoid confusion. Thanks for the fast feedback.